File: demo/lang/go/chall.go
package main

import "fmt"

func main ()  {
	fmt.Println("Input the flag:")
	var flag string
	_, err := fmt.Scanf("%s", &flag)
	if err != nil {
		fmt.Println("Input error!")
		return
	}
	slice_flag := []byte(flag)
	for i := 0; i < len(slice_flag); i++{
		slice_flag[i] ^= byte(0x10 + i)
	}
	cipher := []byte("v}storyHtxt|i|yz]")
	if len(slice_flag) != len(cipher){
		fmt.Println("Wrong")
		return
	}
	for i := 0; i < len(cipher); i++{
		if cipher[i] != slice_flag[i]{
			fmt.Println("Wrong")
			return
		}
	}
	fmt.Println("Good")
	return
}

File: demo/lang/java/Chall.java
import java.util.Scanner;

public class Chall {
    public static void main(String[] args) {
        System.out.println("Input your flag:");
        Scanner scan = new Scanner(System.in);
        String flag = scan.nextLine();
        byte f[] = flag.getBytes();
        int res = 0;
        byte cipher[] = {0x66, 0x6d, 0x63, 0x64, 0x7f, 0x4f, 0x67, 0x71, 0x69, 0x56, 0x66, 0x6a, 0x62, 0x6a, 0x7b, 0x6e, 0x77, 0x74, 0x6f};
        scan.close();
        if (f.length != cipher.length) {
            System.out.println("Wrong");
            return;
        }
        for (int i = 0; i < f.length; i++) {
            f[i] ^= i;
            res |= f[i] ^ cipher[i];
        }
        if (res != 0) {
            System.out.println("Wrong");
        } else {
            System.out.println("Good");
        }
    }
}
